pyproject.toml

```diff
@@ -62,6 +62,8 @@ httpx = "0.27.2"
 mashumaro = "3.13.1"
 msal = "1.31.0"
 python = ">=3.11,<3.13"
+anyio = "4.4.0"
+requests = "2.32.2"
```

Comment:
Just curious: can the above `poetry.lock` changes be made without pinning these dependencies here? On the Node side, that's effectively what we do.

Reason being: it can be a bit confusing to add these dependencies for a bunch of packages that the SDK itself doesn't actually use. Someone may come by later and look at these and find they're not actually used in the project and remove them; would that cause a problem?

CC: @bachya

Reply:
That is a great consideration. These updates are to attempt to remediate the fact that a few of the other dependencies are still relying on insecure versions of those two specific packages. This was the recommended approach by our security tooling on how to properly attempt to mitigate them until the main dependencies correct them. 

Would it help to capture them as a sub-section via comments or something that helps highlight why the are there?

Reply:
Comments sound like they'd be helpful, yeah! I'd defer entirely to @bachya, though, as I know close to nothing about Python package management 😄

In Node-land we don't even have the option of adding comments since `package.json` uses standard JSON format.. which I think is a big reason why I pushed for not including them in the deps list there

With all that said I'd also be totally fine moving forward as is, just wanted to bring up the consideration

Reply:
At a minimum, I would recommend we do something like:

```
anyio = ">=4.4.0"
requests = ">=2.32.2"
```

This will allow Renovate to automatically bump future versions without impact to `pyproject.toml`.

@timdavish To your question, this is one area where Poetry (and Python packaging as a whole) is behind NPM: we don't have a way to specify transitive packages without `pyproject.toml`.

@jayliminal Comments are supported in TOML. 👍🏻

Reply:
Comments added and updated sub-dependency versioning including `>=`.